### PR TITLE
[Snyk] Security upgrade jest from 26.4.2 to 27.0.2

### PR DESCRIPTION
**Pull request by esarafianou**
snyk-top-banner

#### Snyk changed the following file(s):

- `webapp/package.json`
- `webapp/package-lock.json`

#### Vulnerabilities that will be fixed with an upgrade:

|  | Issue | Score | 
:-------------------------:|:-------------------------|:-------------------------
high severity | Regular Expression Denial of Service (ReDoS) <br/>[SNYK-JS-CROSSSPAWN-8303230](https://snyk.io/vuln/SNYK-JS-CROSSSPAWN-8303230) | &nbsp;&nbsp;**756**&nbsp;&nbsp; 
high severity | Uncontrolled resource consumption <br/>[SNYK-JS-BRACES-6838727](https://snyk.io/vuln/SNYK-JS-BRACES-6838727) | &nbsp;&nbsp;**696**&nbsp;&nbsp; 
medium severity | Missing Release of Resource after Effective Lifetime <br/>[SNYK-JS-INFLIGHT-6095116](https://snyk.io/vuln/SNYK-JS-INFLIGHT-6095116) | &nbsp;&nbsp;**631**&nbsp;&nbsp; 
high severity | Inefficient Regular Expression Complexity <br/>[SNYK-JS-MICROMATCH-6838728](https://snyk.io/vuln/SNYK-JS-MICROMATCH-6838728) | &nbsp;&nbsp;**589**&nbsp;&nbsp; 
high severity | Prototype Pollution <br/>[SNYK-JS-UNSETVALUE-2400660](https://snyk.io/vuln/SNYK-JS-UNSETVALUE-2400660) | &nbsp;&nbsp;**589**&nbsp;&nbsp; 

---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: 
🧐 [View latest project report](https://app.snyk.io/org/mattermost-core/project/d292614a-6fd9-4005-81e9-6016aee89e37?utm_source&#x3D;github-enterprise&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates)
🛠 [Adjust project settings](https://app.snyk.io/org/mattermost-core/project/d292614a-6fd9-4005-81e9-6016aee89e37?utm_source&#x3D;github-enterprise&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Uncontrolled resource consumption](https://learn.snyk.io/lesson/redos/?loc&#x3D;fix-pr)
🦉 [Prototype Pollution](https://learn.snyk.io/lesson/prototype-pollution/?loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"jest","from":"26.4.2","to":"27.0.2"}],"env":"prod","issuesToFix":["SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-BRACES-6838727","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-INFLIGHT-6095116","SNYK-JS-INFLIGHT-6095116","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-MICROMATCH-6838728","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660","SNYK-JS-UNSETVALUE-2400660"],"prId":"17dabdac-0678-420e-b144-ac47fab23a22","prPublicId":"17dabdac-0678-420e-b144-ac47fab23a22","packageManager":"npm","priorityScoreList":[696,756,631,589,589],"projectPublicId":"d292614a-6fd9-4005-81e9-6016aee89e37","projectUrl":"https://app.snyk.io/org/mattermost-core/project/d292614a-6fd9-4005-81e9-6016aee89e37?utm_source=github-enterprise&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["updated-fix-title","priorityScore"],"type":"auto","upgrade":["SNYK-JS-BRACES-6838727","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-INFLIGHT-6095116","SNYK-JS-MICROMATCH-6838728","SNYK-JS-UNSETVALUE-2400660"],"vulns":["SNYK-JS-BRACES-6838727","SNYK-JS-CROSSSPAWN-8303230","SNYK-JS-INFLIGHT-6095116","SNYK-JS-MICROMATCH-6838728","SNYK-JS-UNSETVALUE-2400660"],"patch":[],"isBreakingChange":true,"remediationStrategy":"vuln"}'

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -76,7 +76,7 @@
         "eslint-plugin-react": "7.20.6",
         "eslint-plugin-react-hooks": "4.1.2",
         "file-loader": "6.1.0",
-        "jest": "26.4.2",
+        "jest": "27.0.2",
```
Comment (raghavaggarwal2308) [inline webapp/package.json:79]: Why have we kept it in both "devDependencies" and "dependencies"?

Comment (wiggin77) [reply webapp/package.json:79]: This was done by a security bot.  I don't think it knows to remove redundant dependencies.

**Comment by raghavaggarwal2308** [comment]
@wiggin77 Sure, we will look into it

**Comment by Kshitij-Katiyar** [comment]
> @raghavaggarwal2308 can you fix the PR to pass CI?

Hey @wiggin77, this is fixed

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   Thanks @raghavaggarwal2308

**Pull request merged**